integration_test/test_xgboost.py

```diff
@@ -54,28 +123,52 @@ def _verify_parameter_logging(self, run, params):
           assert params[p] == run.metadata[p]
         else:
           assert str(params[p]) == run.metadata[p]
+    assert run.assignments['num_boost_round'] == self.run_params['num_boost_round']
 
-  def _verify_metric_logging(self, run, params):
-    pass
+  def _verify_metric_logging(self, run):
+    data_names = ['Training Set']
+    if self.run_params['evals']:
+      data_names.extend([e[-1] for e in self.run_params['evals']])
+    if self.is_classification:
+      for d_name, m_name in itertools.product(data_names, CLASSIFICATION_METRIC_NAMES):
+        assert 0 <= run.values['-'.join((d_name, m_name))].value <= 1
+    else:
+      for d_name, m_name in itertools.product(data_names, REGRESSION_METRIC_NAMES):
+        assert run.values['-'.join((d_name, m_name))].value >= 0
 
-  def test_run(self):
-    xgb_params = _form_random_run_params()
-    ctx = sigopt.xgboost.run(**xgb_params)
-    run = sigopt.get_run(ctx.run.id)
-    assert run.metadata['Dataset columns'] == 4
-    assert run.metadata['Dataset rows'] == 120
-    assert run.metadata['Number of Test Sets'] == 1
+    if self.run_params['params']['eval_metric']:
+      for d_name, m_name in itertools.product(data_names[1:], self.run_params['params']['eval_metric']):
+        assert run.values['-'.join((d_name, m_name))]
+
+    assert run.values['Training time'].value > 0
+
+
+  def _verify_metadata_logging(self, run):
+    assert run.metadata['Dataset columns'] == self.run_params['dtrain'].num_col()
+    assert run.metadata['Dataset rows'] == self.run_params['dtrain'].num_row()
+    assert run.metadata['Number of Test Sets'] == len(self.run_params['evals'])
     assert run.metadata['Python Version'] == platform.python_version()
     assert run.metadata['XGBoost Version'] == xgb.__version__
-    self._verify_parameter_logging(run, xgb_params['params'])
-    assert run.assignments['num_boost_round'] == xgb_params['num_boost_round']
+
+  def test_run(self):
+    self._form_random_run_params()
```

Comment:
I prefer for each type of model, add a new test case. In each case, there should be several random samples.

Reply:
I see what you are saying and I agree in general. However, I want to wait till I know how long this test takes on circleci before deciding. If each test (xgb.train) takes <30 seconds, then sure let's split this into one test for each task. 

Mostly, I just want to avoid having people waiting for this test to finish every time.

Reply:
I think one small model for each model type is OK.

Reply:
Cool. I'll make the change.